addons/crm/wizard/base_partner_merge.py

```diff
@@ -0,0 +1,608 @@
+        """
+        Start the process.
+        * Compute the selected groups (with duplication)
+        * If the user has selected the 'exclude_XXX' fields, avoid the partners.
+        """
+        self.ensure_one()
+        groups = self._compute_selected_groupby()
+        query = self._generate_query(groups, self.maximum_group)
+        self.with_context(active_test=False)._process_query(query)
+        return self.with_context(active_test=False)._next_screen()
+
+    @api.multi
+    def update_all_process_cb(self):
+        self.ensure_one()
+        self.parent_migration_process_cb()
+
+        list_merge = [
+            {'group_by_vat': True, 'group_by_email': True, 'group_by_name': True},
+        ]
+
+        for merge_value in list_merge:
+            rec = self.create(merge_value)
+            rec.automatic_process_cb()
+
+        self.env.cr.execute("""
+            UPDATE
+                res_partner
+            SET
+                is_company = NULL
+            WHERE
+                parent_id IS NOT NULL AND
+                is_company IS NOT NULL
+        """)
+
+        return self._next_screen()
+
+    @api.multi
+    def next_cb(self):
+        """
+        Don't compute any thing
+        """
+        self.ensure_one()
+        self.with_context(active_test=False).current_line_id.unlink()
+        return self.with_context(active_test=False)._next_screen()
+
+    def _get_fk_on(self, table):
+        q = """  SELECT cl1.relname as table,
+                        att1.attname as column
+                   FROM pg_constraint as con, pg_class as cl1, pg_class as cl2,
+                        pg_attribute as att1, pg_attribute as att2
+                  WHERE con.conrelid = cl1.oid
+                    AND con.confrelid = cl2.oid
+                    AND array_lower(con.conkey, 1) = 1
+                    AND con.conkey[1] = att1.attnum
+                    AND att1.attrelid = cl1.oid
+                    AND cl2.relname = %s
+                    AND att2.attname = 'id'
+                    AND array_lower(con.confkey, 1) = 1
+                    AND con.confkey[1] = att2.attnum
+                    AND att2.attrelid = cl2.oid
+                    AND con.contype = 'f'
+        """
+        return self.env.cr.execute(q, (table,))
+
+    @api.model
+    def _update_foreign_keys(self, src_partners, dst_partner):
+        _logger.debug('_update_foreign_keys for dst_partner: %s for src_partners: %r', dst_partner.id, list(map(operator.attrgetter('id'), src_partners)))
+
+        # find the many2one relation to a partner
+        Partner = self.env['res.partner']
+        self._get_fk_on('res_partner')
+
+        # ignore two tables
+
+        for table, column in self.env.cr.fetchall():
+            if 'base_partner_merge_' in table:
+                continue
+            partner_ids = tuple(map(int, src_partners))
+
+            query = "SELECT column_name FROM information_schema.columns WHERE table_name LIKE '%s'" % (table)
```

Comment:
@WHPODOO 

Here we can't able to use information_schema in search method because information_schema manage by a PostgreSQL.